src/Workspaces/Core/Portable/Execution/Serialization/HierarchicalChecksumObject.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Execution
+{
+    /// <summary>
+    /// HierarchicalChecksumObject indicates this type is collection of checksumObject types.
```

Comment:
it is marker type. showing this sub type is all hierarchical checksum object. so that I can restrict input or out to certain type.